Sources/Testing/Events/Event.swift

```diff
@@ -252,3 +252,148 @@ extension Event {
+    }
+  }
+}
+
+extension Event.Kind {
+  /// A serializable enumeration describing the various kinds of event that can be observed.
+  @_spi(ExperimentalSnapshotting)
+  public enum Snapshot: Sendable, Codable {
+    /// A test run started.
+    ///
+    /// This is the first event posted after ``Runner/run()`` is called.
+    @_spi(ExperimentalTestRunning)
+    case runStarted
+
+    /// A step in the runner plan started.
+    ///
+    /// - Parameters:
+    ///   - step: The step in the runner plan which started.
+    ///
+    /// This is posted when a ``Runner`` begins processing a
+    /// ``Runner/Plan/Step``. Processing this step may result in its associated
+    /// ``Test`` being run, skipped, or another action, so this event will only
+    /// be followed by a ``testStarted`` event if the step's test is run.
+    @_spi(ExperimentalTestRunning)
+    case planStepStarted
+
+    /// A test started.
+    case testStarted
+
+    /// A test case started.
+    @_spi(ExperimentalParameterizedTesting)
+    case testCaseStarted
+
+    /// A test case ended.
+    @_spi(ExperimentalParameterizedTesting)
+    case testCaseEnded
+
+    /// An expectation was checked with `#expect()` or `#require()`.
+    ///
+    /// - Parameters:
+    ///   - expectation: The expectation which was checked.
+    ///
+    /// By default, events of this kind are not generated because they occur
+    /// frequently in a typical test run and can generate significant
+    /// backpressure on the event handler.
+    ///
+    /// Failed expectations also, unless expected to fail, generate events of
+    /// kind ``Event/Kind-swift.enum/issueRecorded(_:)``. Those events are
+    /// always posted to the current event handler.
+    ///
+    /// To enable events of this kind, set
+    /// ``Configuration/deliverExpectationCheckedEvents`` to `true` before
+    /// running tests.
+    case expectationChecked(_ expectation: Expectation.Snapshot)
+
+    /// An issue was recorded.
+    ///
+    /// - Parameters:
+    ///   - issue: The issue which was recorded.
+    case issueRecorded(_ issue: Issue.Snapshot)
+
+    /// A test ended.
+    case testEnded
+
+    /// A test was skipped.
+    ///
+    /// - Parameters:
+    ///   - skipInfo: A ``SkipInfo`` containing details about this skipped test.
+    case testSkipped(_ skipInfo: SkipInfo)
+
+    /// A step in the runner plan ended.
+    ///
+    /// - Parameters:
+    ///   - step: The step in the runner plan which ended.
+    ///
+    /// This is posted when a ``Runner`` finishes processing a
+    /// ``Runner/Plan/Step``.
+    @_spi(ExperimentalTestRunning)
+    case planStepEnded
+
+    /// A test run ended.
+    ///
+    /// This is the last event posted before ``Runner/run()`` returns.
+    @_spi(ExperimentalTestRunning)
+    case runEnded
+
+    /// Snapshots an ``Event.Kind``.
+    /// - Parameter kind: The original ``Event.Kind`` to snapshot.
+    @_spi(ExperimentalSnapshotting)
+    init(snapshotting event: Event.Kind) {
```

Comment:
```suggestion
    public init(snapshotting event: Event.Kind) {
```

And they shouldn't need `@_spi` since the containing declaration already has it

Reply:
Wow, thanks! Too early in the morning ...